conans/client/build/meson.py

```diff
@@ -151,15 +152,19 @@ def configure(self, args=None, defs=None, source_dir=None, build_dir=None,
             build_type
         ])
         command = 'meson "%s" "%s" %s' % (source_dir, self.build_dir, arg_list)
-        command = self._append_vs_if_needed(command)
         with tools.environment_append({"PKG_CONFIG_PATH": pc_paths}):
-            self._conanfile.run(command)
+            self._run(command)
 
-    def _append_vs_if_needed(self, command):
-        if self._compiler == "Visual Studio" and self.backend == "ninja":
-            vcvars = tools.vcvars_command(self._conanfile.settings, output=self._conanfile.output)
-            command = "%s && %s" % (vcvars, command)
-        return command
+    @property
+    def _vcvars_needed(self):
+        return self._compiler == "Visual Studio" and self.backend == "ninja"
+
+    def _run(self, command):
+        with tools.vcvars(self._settings,
+                          output=self._conanfile.output) if self._vcvars_needed else tools.no_op():
+            env_build = AutoToolsBuildEnvironment(self._conanfile)
```

Comment:
At first sight it doesn't feel great to use another build helper and especially that one based only on the env vars.
Is there another thing we can do like write a file to be injected to meson?

Reply:
seems like writing to file is something we will need to solve https://github.com/conan-io/conan/issues/4529
but I think it's blocked by https://github.com/conan-io/conan/pull/5184 
we need to do some design decisions on how to handle such kind of files, are they produced by their own generators, or by existing generators, or some other entities.
I think right now it's too much for the original particular problem.
I prefer to go by small iterations, like first solve flags issue, then solve cross-compiling issue, and then solve toolchain file design problems. I think it's too much to do at single shot.
for the `AutoToolsBuildEnvironment`, why not re-use it, at least for the first iteration? as soon as meson is compatible with auto-tools variables, we can re-use the code, especially as it's already well-tested and proven to work correctly.